modules/siddhi-extensions/output-mappers/json-output-mapper/src/main/java/org/wso2/siddhi/extension/output/mapper/json/JsonSinkMapper.java

```diff
@@ -0,0 +1,361 @@
+            throws ConnectionUnavailableException {
+        StringBuilder sb = null;
+        if (payloadTemplateBuilder == null) {
+            String jsonString = constructJsonForDefaultMapping(events);
+            if (jsonString != null) {
+                sb = new StringBuilder();
+                sb.append(jsonString);
+            }
+        } else {
+            sb = new StringBuilder();
+            sb.append(constructJsonForCustomMapping(events, payloadTemplateBuilder));
+        }
+
+        if (sb != null) {
+            if (!isJsonValidationEnabled) {
+                sinkListener.publish(sb.toString(), dynamicOptions);
+            } else if (isValidJson(sb.toString())) {
+                sinkListener.publish(sb.toString(), dynamicOptions);
+            } else {
+                log.error("Invalid json string : " + sb.toString() + ". Hence dropping the message.");
+            }
+        }
+    }
+
+    @Override
+    public void mapAndSend(Event event, OptionHolder optionHolder, TemplateBuilder payloadTemplateBuilder,
+                           SinkListener sinkListener, DynamicOptions dynamicOptions)
+            throws ConnectionUnavailableException {
+        StringBuilder sb = null;
+        if (payloadTemplateBuilder == null) {
+            String jsonString = constructJsonForDefaultMapping(event);
+            if (jsonString != null) {
+                sb = new StringBuilder();
+                sb.append(jsonString);
+            }
+        } else {
+            sb = new StringBuilder();
+            sb.append(constructJsonForCustomMapping(event, payloadTemplateBuilder));
+        }
+
+        if (sb != null) {
+            if (!isJsonValidationEnabled) {
+                sinkListener.publish(sb.toString(), dynamicOptions);
+            } else if (isValidJson(sb.toString())) {
+                sinkListener.publish(sb.toString(), dynamicOptions);
+            } else {
+                log.error("Invalid json string : " + sb.toString() + ". Hence dropping the message.");
+            }
+        }
+    }
+
+    private String constructJsonForDefaultMapping(Object eventObj) {
+        StringBuilder sb = new StringBuilder();
+        int numberOfOuterObjects = 0;
+        if (enclosingElement != null) {
+            String[] nodeNames = enclosingElement.split("\\.");
+            if (DEFAULT_ENCLOSING_ELEMENT.equals(nodeNames[0])) {
+                numberOfOuterObjects = nodeNames.length - 1;
+            } else {
+                numberOfOuterObjects = nodeNames.length;
+            }
+            for (String nodeName : nodeNames) {
+                if (!DEFAULT_ENCLOSING_ELEMENT.equals(nodeName)) {
+                    sb.append(JSON_EVENT_START_SYMBOL).append("\"").append(nodeName).append("\"")
+                            .append(JSON_KEYVALUE_SEPERATOR);
+                }
+            }
+            if (eventObj.getClass() == Event.class) {
```

Comment:
It was discussed in the code review to use instance of because Event may get extended in future